python/opae.admin/opae/admin/tools/regmap_debugfs.py

```diff
@@ -0,0 +1,149 @@
+"""
+Copyright(c) 2020, Intel Corporation
+
+Redistribution  and  use  in source  and  binary  forms,  with  or  without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of  source code  must retain the  above copyright notice,
+  this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+* Neither the name  of Intel Corporation  nor the names of its contributors
+  may be used to  endorse or promote  products derived  from this  software
+  without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+"""
+
+import argparse
+import os.path
+
+
+class RegmapDebugfs():
+    """RegmapDebugfs is a class for accessing the debugfs
+       inteface of regmaps"""
+
+    def __init__(self, regmapdir):
+        if not os.path.isdir(regmapdir):
+            raise Exception("bad regmap dir")
+
+        _range = os.path.join(regmapdir, 'range')
+        if not os.path.isfile(_range):
+            raise Exception("could not find range entry")
+
+        _registers = os.path.join(regmapdir, 'registers')
+        if not os.path.isfile(_registers):
+            raise Exception("could not find registers entry")
+
+        self._registers = _registers
+
+        with open(self._registers, 'r') as _file:
+            line0 = _file.readline()
```

Comment:
Is there better description of what this line is doing? `line0` and `line1` are not very straightforward, and do not explain the purpose of this.

Reply:
The idea here is to read two lines from the debugfs and compare the addresses to get the stride between registers.  In some cases the addresses are word address, and in other cases the addresses are byte addresses.  I'll add comments and make better names.